Fix the remaining Clang warnings

A list of interesting warnings that we probably want to fix (not necessarily complete because sifting through the logs is hard):
```
./quantities/parser_body.hpp:24:13: warning: suggest braces around initialization of subobject [-Wmissing-braces]
    return {0, 0, 0, 0, 0, 0, 0, 0};
            ^~~~~~~~~~~~~~~~~~~~~~
            {                     }
./quantities/parser_body.hpp:46:13: warning: suggest braces around initialization of subobject [-Wmissing-braces]
    return {LengthExponent,
            ^~~~~~~~~~~~~~~
astronomy/ksp_system_test.cpp:502:43: warning: must specify at least one argument for '...' parameter of variadic macro [-Wgnu-zero-variadic-macro-arguments]
            /*first_step_in_seconds=*/75}));
astronomy/solar_system_dynamics_test.cpp:581:43: warning: must specify at least one argument for '...' parameter of variadic macro [-Wgnu-zero-variadic-macro-arguments]
            /*first_step_in_seconds=*/75}));
                                          ^
integrators/symplectic_runge_kutta_nyström_integrator_test.cpp:724:46: warning: must specify at least one argument for '...' parameter of variadic macro [-Wgnu-zero-variadic-macro-arguments]
                        ValuesIn(Instances()));
integrators/symmetric_linear_multistep_integrator_test.cpp:452:46: warning: must specify at least one argument for '...' parameter of variadic macro [-Wgnu-zero-variadic-macro-arguments]
                        ValuesIn(Instances()));
physics/ephemeris_test.cpp:1021:59: warning: must specify at least one argument for '...' parameter of variadic macro [-Wgnu-zero-variadic-macro-arguments]
        &Quinlan1999Order8A<Position<ICRFJ2000Equator>>()));
./base/array_body.hpp:48:13: warning: suggest braces around initialization of subobject [-Wmissing-braces]
    : data_{std::forward<Args>(args)...},
            ^~~~~~~~~~~~~~~~~~~~~~~~
            {                       }
ksp_plugin_test/interface_renderer_test.cpp:49:11: warning: unused variable 'trajectory_size' [-Wunused-const-variable]
int const trajectory_size = 10;
          ^
ksp_plugin_test/interface_renderer_test.cpp:51:5: warning: unused variable 'parent_position' [-Wunused-variable]
XYZ parent_position = {4, 5, 6};
    ^
./base/array_body.hpp:48:13: warning: suggest braces around initialization of subobject [-Wmissing-braces]
    : data_{std::forward<Args>(args)...},
            ^~~~~~~~~~~~~~~~~~~~~~~~
            {                       }
```

--- a/base/array_body.hpp
+++ b/base/array_body.hpp
@@ -45,7 +45,7 @@ Array<Element> UniqueArray<Element>::get() const {
 template<typename Element, std::int32_t max_size>
 template<typename... Args>
 constexpr BoundedArray<Element, max_size>::BoundedArray(Args&&... args)
-    : data_{std::forward<Args>(args)...},
+    : data_{{std::forward<Args>(args)...}},
       size_(sizeof...(args)) {}
 
 template<typename Element, std::int32_t max_size>

--- a/ksp_plugin_test/interface_renderer_test.cpp
+++ b/ksp_plugin_test/interface_renderer_test.cpp
@@ -46,10 +46,6 @@ Index const celestial_index = 1;
 Index const parent_index = 2;
 Index const unused = 666;
 
-int const trajectory_size = 10;
-
-XYZ parent_position = {4, 5, 6};
-
 }  // namespace
 
 class InterfaceRendererTest : public ::testing::Test {

--- a/quantities/parser_body.hpp
+++ b/quantities/parser_body.hpp
@@ -21,7 +21,7 @@ struct ExtractDimensions {};
 template<>
 struct ExtractDimensions<double> {
   static constexpr RuntimeDimensions dimensions() {
-    return {0, 0, 0, 0, 0, 0, 0, 0};
+    return {{0, 0, 0, 0, 0, 0, 0, 0}};
   }
 };
 
@@ -43,14 +43,14 @@ struct ExtractDimensions<
                                              LuminousIntensityExponent,
                                              AngleExponent>>> {
   static constexpr RuntimeDimensions dimensions() {
-    return {LengthExponent,
-            MassExponent,
-            TimeExponent,
-            CurrentExponent,
-            TemperatureExponent,
-            AmountExponent,
-            LuminousIntensityExponent,
-            AngleExponent};
+    return {{LengthExponent,
+             MassExponent,
+             TimeExponent,
+             CurrentExponent,
+             TemperatureExponent,
+             AmountExponent,
+             LuminousIntensityExponent,
+             AngleExponent}};
   }
 };
 

